Cesium3DTilesSelection/src/TilesetContentManager.cpp

```diff
@@ -1722,16 +1759,24 @@ void TilesetContentManager::setTileContent(
     }
 
     auto& content = tile.getContent();
+    CESIUM_ASSERT(content.isUnknownContent());
+
     std::visit(
         ContentKindSetter{
             content,
             std::move(result.rasterOverlayDetails),
             pWorkerRenderResources},
         std::move(result.contentKind));
 
-    if (!tile.getContent().isUnknownContent()) {
-      tile.incrementDoNotUnloadSubtreeCountOnParent(
-          "TilesetContentManager::setTileContent");
+    // "Unknown" content is not loaded, and does not get a reference.
+    // Also, if a tile ID is not loadable, it can never be unloaded (because how
+    // would we reload it?) and so that doesn't get a reference, either.
+    // An example of non-loadable content is the "external content" at the root
+    // of a regular tileset.json tileset. We can't unload it without destroying
+    // the entire tileset.
+    if (!tile.getContent().isUnknownContent() &&
+        TileIdUtilities::isLoadable(tile.getTileID())) {
```

Comment:
Not a hard requirement, but I saw this check a lot throughout the code and wonder if it's worth making a helper / member function on the tile. Like `hasReferenceableContent` or something. And having this comment on the function would be a helpful doc